src/lunamary/ReadWriteData/AbstractFactory.java

```diff
@@ -0,0 +1,20 @@
+package lunamary.ReadWriteData;
+
+public class AbstractFactory {
+
+    public ReadWriteFile createFile(String path) {
```

Comment:
Factory methods are usually static so there is no need to instantiate objects of these classes to make use of them